src/htm/algorithms/TemporalMemory.cpp

```diff
@@ -218,7 +186,7 @@ static void growSynapses(Connections &connections,
   vector<CellIdx> candidates(prevWinnerCells.begin(), prevWinnerCells.end());
   NTA_ASSERT(std::is_sorted(candidates.begin(), candidates.end()));
 
-  // Remove cells that are already synapsed on by this segment
+  // Skip cells that are already synapsed on by this segment //TODO is this biological? Randomly creating a synapse would be faster! 
```

Comment:
> is this biological?

I think it is.  There are structural constraints on the shapes of axons & synapses which prevent a large number duplicate of connections.

I think it's important to prevent cells from growing duplicate synapses from growing onto a segment, because otherwise a strong input will be sampled many times and grow many synapses, which gives that input a stronger connection.  Synapses are supposed to have binary effects (0 or 1) but duplicate synapses give them varying levels of strength.